kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -403,7 +411,13 @@ public WriteModel<WindowDoc> delete(
       final int kafkaPartition,
       final WindowedKey windowedKey
   ) {
+    if (retainDuplicates) {
+      throw new IllegalStateException("Should never attempt to issue a delete with duplicates");
```

Comment:
(not a blocker) why? couldn't the user have a custom processor that does this and expects it to delete all the values associated with the same key?

Reply:
I mean...they can expect that, and it does make sense, but it's not how the built-in stores in Kafka Streams work so I figured we should follow the existing semantics instead of correcting them

In general the window store with duplicates isn't very well defined or thought out since it was never really supposed to be for users to plug into their processors but rather a hack to allow reusing the WindowStore for the new (at the time) stream-stream join operator. We don't explicitly forbid anyone from trying to use this but there's not much interest out there. In the end deletes are sort of like point lookups, in that the API is built around the non-duplicate stores and just happened to be inherited without any kind of semantic guarantees.

Reply:
Btw we throw an exception here because the delete should be "stopped" in one of the above layers like the ResponsiveWindowStore, so the call should never reach the MongoWindowedTable. ie it's not that it's invalid for a user to attempt a delete (the store just treats it as a no-op), rather it's invalid for our code to attempt this since we expect deletes to be intercepted before this